app/test/process_proposal_test.go

```diff
@@ -328,6 +328,23 @@ func TestProcessProposal(t *testing.T) {
 			appVersion:     v3.Version,
 			expectedResult: abci.ResponseProcessProposal_REJECT,
 		},
+		{
+			name:  "tx size exceeds max tx size limit",
+			input: validData(),
+			mutator: func(d *tmproto.Data) {
+				maxTxSize := appconsts.MaxTxSize(testApp.AppVersion()) // max tx size for the latest version
+				// set the blob size to maxTxSize so that the raw transaction size will exceeds the max tx size limit
+				blob, err := share.NewBlob(ns1, bytes.Repeat([]byte{1}, maxTxSize), appconsts.DefaultShareVersion, nil)
+				require.NoError(t, err)
+				rawTx, _, err := signer.CreatePayForBlobs(accounts[0], []*share.Blob{blob}, user.SetGasLimit(100000), user.SetFee(100000))
+				require.NoError(t, err)
+				// override the last valid blob tx with large one that exceeds the max tx size limit
+				// proposal block should be rejected
+				d.Txs[2] = rawTx
+			},
+			appVersion:     v3.Version,
```

Comment:
[blocking] this shouldn't apply to v3 because it's breaking. It should only apply to v4.